geoutils/georaster/raster.py

```diff
@@ -229,69 +310,92 @@ def __init__(
         else:
             raise ValueError("filename argument not recognised.")
 
-        self._read_attrs(attrs)
+    @property
+    def nbands(self) -> int:
+        if not self.is_loaded and self._disk_shape is not None:
+            return self._disk_shape[0]
+        return int(self.data.shape[0])
 
-        # Save _masked attribute to be used by self.load()
-        self._masked = masked
+    @property
+    def count(self) -> int:
+        if self._disk_shape is not None:
+            return self._disk_shape[0]
+        return self.nbands
 
-        # Check number of bands to be loaded
-        if bands is None:
-            nbands = self.count
-        elif isinstance(bands, int):
-            nbands = 1
-        elif isinstance(bands, abc.Iterable):
-            nbands = len(bands)
-
-        # Downsampled image size
-        if not isinstance(downsample, (int, float)):
-            raise ValueError("downsample must be of type int or float")
-        if downsample == 1:
-            out_shape = (nbands, self.height, self.width)
-        else:
-            down_width = int(np.ceil(self.width / downsample))
-            down_height = int(np.ceil(self.height / downsample))
-            out_shape = (nbands, down_height, down_width)
-
-        if load_data:
-            self.load(bands=bands, out_shape=out_shape)
-            self.nbands = self._data.shape[0]
-            self.is_loaded = True
-            if isinstance(filename_or_dataset, str):
-                self._is_modified = False
-                self._disk_hash = hash((self._data.tobytes(), self.transform, self.crs, self.nodata))
-        else:
-            self._data = None
-            self.nbands = None
-            self.is_loaded = False
+    @property
+    def height(self) -> int:
+        if not self.is_loaded:
+            return self._disk_shape[1]  # type: ignore
+        return int(self.data.shape[1])
 
-        # update attributes when downsample is not 1
-        if downsample != 1:
+    @property
+    def width(self) -> int:
+        if not self.is_loaded:
+            return self._disk_shape[2]  # type: ignore
+        return int(self.data.shape[2])
 
-            # Original attributes
-            meta = self.ds.meta
+    @property
+    def shape(self) -> tuple[int, int]:
+        if not self.is_loaded:
+            return self._disk_shape[1], self._disk_shape[2]  # type: ignore
+        return int(self.data.shape[1]), int(self.data.shape[2])
+
+    @property
+    def res(self) -> tuple[float | int, float | int]:
+        return self.transform[0], abs(self.transform[4])
+
+    @property
+    def bounds(self) -> rio.coords.BoundingBox:
+        left = self.transform[2]
+        top = self.transform[5]
+        res = self.res
+        return rio.coords.BoundingBox(
```

Comment:
It would be more straightforward to use this: https://rasterio.readthedocs.io/en/latest/api/rasterio.transform.html#rasterio.transform.array_bounds

Reply:
Implemented in def1811